arbor/mechcat.cpp

```diff
@@ -131,18 +166,34 @@ std::unique_ptr<mechanism> mechanism_catalogue::instance_impl(std::type_index ti
         }
     }
 
-    std::unique_ptr<mechanism> mech = prototype->clone();
+    mech.first = prototype->clone();
 
-    auto apply_globals = [this](auto& self, const std::string& name, mechanism* mptr) -> void {
+    auto apply_globals = [this](auto& self, const std::string& name, mechanism_overrides& over) -> void {
```

Comment:
This lambda has overflowed its boundaries somewhat. Might be worth making it a member function.

Reply:
Doing so just feels like I'm splitting a method in two; will do it if you insist, but I don't think it's going to be a big gain in clarity.

Reply:
I see now that this was done to enable recursion.
A little bit of documentation that explains the algorithm or approach being taken here would make it much easier to understand.

Reply:
This could be replaced with e.g. a stack/vector and a for loop, if that would make more sense.